### PR TITLE
[Cinematic] Add support for Xcode 15 beta 6.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/rsp/dotnet/ios-defines-dotnet.rsp [hunk 1]
```diff
@@ -22,6 +22,7 @@
 -d:HAS_CALLKIT
 -d:HAS_CARPLAY
 -d:HAS_CFNETWORK
+-d:HAS_CINEMATICS
```
Comment (mandel-macaque) [inline src/rsp/dotnet/ios-defines-dotnet.rsp:25]:
```suggestion
-d:HAS_CINEMATIC
```

File: src/rsp/dotnet/maccatalyst-defines-dotnet.rsp [hunk 2]
```diff
@@ -19,6 +19,7 @@
 -d:HAS_BUSINESSCHAT
 -d:HAS_CALLKIT
 -d:HAS_CFNETWORK
+-d:HAS_CINEMATICS
```
Comment (mandel-macaque) [inline src/rsp/dotnet/maccatalyst-defines-dotnet.rsp:22]:
```suggestion
-d:HAS_CINEMATIC
```

File: src/rsp/dotnet/macos-defines-dotnet.rsp [hunk 3]
```diff
@@ -16,6 +16,7 @@
 -d:HAS_BUSINESSCHAT
 -d:HAS_CALLKIT
 -d:HAS_CFNETWORK
+-d:HAS_CINEMATICS
```
Comment (mandel-macaque) [inline src/rsp/dotnet/macos-defines-dotnet.rsp:19]:
```suggestion
-d:HAS_CINEMATIC
```

File: src/rsp/dotnet/tvos-defines-dotnet.rsp [hunk 4]
```diff
@@ -9,6 +9,7 @@
 -d:HAS_AVKIT
 -d:HAS_BACKGROUNDTASKS
 -d:HAS_CFNETWORK
+-d:HAS_CINEMATICS
```
Comment (mandel-macaque) [inline src/rsp/dotnet/tvos-defines-dotnet.rsp:12]:
```suggestion
-d:HAS_CINEMATIC
```

File: src/rsp/ios-defines.rsp [hunk 5]
```diff
@@ -23,6 +23,7 @@
 -d:HAS_CARPLAY
 -d:HAS_CFNETWORK
 -d:HAS_CHIP
+-d:HAS_CINEMATICS
```
Comment (mandel-macaque) [inline src/rsp/ios-defines.rsp:26]:
```suggestion
-d:HAS_CINEMATIC
```

File: src/rsp/macos-defines.rsp [hunk 6]
```diff
@@ -17,6 +17,7 @@
 -d:HAS_CALLKIT
 -d:HAS_CFNETWORK
 -d:HAS_CHIP
+-d:HAS_CINEMATICS
```
Comment (mandel-macaque) [inline src/rsp/macos-defines.rsp:20]:
```suggestion
-d:HAS_CINEMATIC
```

File: src/rsp/tvos-defines.rsp [hunk 7]
```diff
@@ -10,6 +10,7 @@
 -d:HAS_BACKGROUNDTASKS
 -d:HAS_CFNETWORK
 -d:HAS_CHIP
+-d:HAS_CINEMATICS
```
Comment (mandel-macaque) [inline src/rsp/tvos-defines.rsp:13]:
```suggestion
-d:HAS_CINEMATIC
```

File: src/Cinematic/CNDecision.cs [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+	[NoWatch, TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+#endif
+	public enum CNDecisionIdentifierType {
+		Single,
+		Group,
+	}
+
+	public partial class CNDecision {
+
+		public CNDecision (CMTime time, long detectionID, bool isStrong, CNDecisionIdentifierType identifierType)
```
Comment (mandel-macaque) [inline src/Cinematic/CNDecision.cs:28]: No need for attr using the default ones from the class.

File: src/Cinematic/CNDecision.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+#nullable enable
+
+#if !WATCH
+
+namespace Cinematic {
+
+#if NET
+	[UnsupportedOSPlatform ("watchos")]
+	[UnsupportedOSPlatform ("tvos17.0")]
```
Comment (mandel-macaque) [inline src/Cinematic/CNDecision.cs:14]:
```suggestion
	[SupportedOSPlatform ("tvos17.0")]
```

File: src/cinematic.cs [hunk 10]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Internal]
+		[Export ("initWithTime:detectionGroupID:strong:")]
+		NativeHandle _InitWithGroupIdentifier (CMTime time, long detectionGroupID, bool isStrong);
+
+		[Export ("time")]
+		CMTime Time { get; }
+
+		[Export ("detectionID")]
+		long DetectionID { get; }
```
Comment (dalexsoto) [inline src/cinematic.cs:245]: Not an acronym 
```suggestion
		long DetectionId { get; }
```

File: src/cinematic.cs [hunk 11]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Export ("detectionGroupID")]
+		long DetectionGroupID { get; }
```
Comment (dalexsoto) [inline src/cinematic.cs:248]:
```suggestion
		long DetectionGroupId { get; }
```

File: src/cinematic.cs [hunk 12]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Static]
+		[Export ("isValidDetectionID:")]
+		bool IsValidDetectionID (long detectionID);
```
Comment (dalexsoto) [inline src/cinematic.cs:214]:
```suggestion
		bool IsValidDetectionId (long detectionId);
```

File: src/cinematic.cs [hunk 13]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Static]
+		[Export ("isValidDetectionGroupID:")]
+		bool IsValidDetectionGroupID (long detectionGroupID);
```
Comment (dalexsoto) [inline src/cinematic.cs:218]:
```suggestion
		bool IsValidDetectionGroupId (long detectionGroupId);
```

File: src/cinematic.cs [hunk 14]
```diff
@@ -0,0 +1,494 @@
...
+
+	[TV (17, 0), NoWatch, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CNDetectionTrack : NSCopying {
+		[Export ("detectionType")]
+		CNDetectionType DetectionType { get; }
+
+		[Export ("detectionID")]
+		long DetectionID { get; }
```
Comment (dalexsoto) [inline src/cinematic.cs:268]:
```suggestion
		long DetectionId { get; }
```

File: src/cinematic.cs [hunk 15]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Export ("detectionGroupID")]
+		long DetectionGroupID { get; }
```
Comment (dalexsoto) [inline src/cinematic.cs:271]:
```suggestion
		long DetectionGroupId { get; }
```

File: src/cinematic.cs [hunk 16]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Export ("focusDetection", ArgumentSemantic.Strong)]
+		CNDetection FocusDetection { get; }
+
+		[Export ("allDetections", ArgumentSemantic.Strong)]
+		CNDetection [] AllDetections { get; }
+
+		[Export ("detectionForID:")]
+		[return: NullAllowed]
+		CNDetection GetDetectionForID (long detectionID);
```
Comment (dalexsoto) [inline src/cinematic.cs:448]:
```suggestion
		CNDetection GetDetectionForId (long detectionId);
```

File: src/cinematic.cs [hunk 17]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Export ("bestDetectionForGroupID:")]
+		[return: NullAllowed]
+		CNDetection GetBestDetectionForGroupID (long detectionGroupID);
```
Comment (dalexsoto) [inline src/cinematic.cs:452]:
```suggestion
		CNDetection GetBestDetectionForGroupId (long detectionGroupId);
```

**Review by dalexsoto — APPROVED** [review]

File: src/cinematic.cs [hunk 18]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Export ("userDecisionsInTimeRange:")]
+		CNDecision [] GetUserDecisions (CMTimeRange timeRange);
+
+		[Export ("baseDecisionsInTimeRange:")]
+		CNDecision [] GetBaseDecisions (CMTimeRange timeRange);
+
+		[Export ("detectionTrackForID:")]
+		[return: NullAllowed]
+		CNDetectionTrack GetDetectionTrackForID (long detectionID);
```
Comment (rolfbjarne) [inline src/cinematic.cs:382]:
```suggestion
		CNDetectionTrack GetDetectionTrackForId (long detectionId);
```

File: src/cinematic.cs [hunk 19]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Export ("initWithCommandQueue:")]
+		NativeHandle Constructor (IMTLCommandQueue commandQueue);
+
+		[Export ("findObjectAtPoint:sourceImage:")]
+		[return: NullAllowed]
+		CNBoundsPrediction FindObject (CGPoint point, CVPixelBuffer sourceImage);
+
+		[Export ("startTrackingAt:within:sourceImage:sourceDisparity:")]
+		bool StartTrackingAt (CMTime time, CGRect normalizedBounds, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity);
```
Comment (rolfbjarne) [inline src/cinematic.cs:481]:
```suggestion
		bool StartTracking (CMTime atTime, CGRect normalizedBounds, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity);
```

File: src/cinematic.cs [hunk 20]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Export ("continueTrackingAt:sourceImage:sourceDisparity:")]
+		[return: NullAllowed]
+		CNBoundsPrediction ContinueTrackingAt (CMTime time, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity);
```
Comment (rolfbjarne) [inline src/cinematic.cs:485]:
```suggestion
		CNBoundsPrediction ContinueTracking (CMTime atTime, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity);
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/cinematic.cs [hunk 21]
```diff
@@ -0,0 +1,494 @@
...
+		CNDetectionType DetectionType { get; }
+
+		[Export ("normalizedRect")]
+		CGRect NormalizedRect { get; }
+
+		[Export ("focusDisparity")]
+		float FocusDisparity { get; }
+
+		[Export ("detectionID")]
+		long DetectionID { get; }
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/cinematic.cs:207]: Dotnet uses ID: https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions and we have more frameworks using ID than Id in our source.

File: src/cinematic.cs [hunk 22]
```diff
@@ -0,0 +1,494 @@
...
+
+		[Export ("detectionGroupID")]
+		long DetectionGroupID { get; }
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/cinematic.cs:210]: Same.

File: src/cinematic.cs (hunk 12 above)
Comment (mandel-macaque) [reply src/cinematic.cs:214]: Same.

File: src/cinematic.cs (hunk 13 above)
Comment (mandel-macaque) [reply src/cinematic.cs:218]: Same

File: src/cinematic.cs (hunk 11 above)
Comment (mandel-macaque) [reply src/cinematic.cs:248]: Same

File: src/cinematic.cs (hunk 10 above)
Comment (mandel-macaque) [reply src/cinematic.cs:245]: Same.

File: src/cinematic.cs (hunk 15 above)
Comment (mandel-macaque) [reply src/cinematic.cs:271]: Same.

File: src/cinematic.cs (hunk 18 above)
Comment (mandel-macaque) [reply src/cinematic.cs:382]: Same.

File: src/cinematic.cs (hunk 16 above)
Comment (mandel-macaque) [reply src/cinematic.cs:448]: Same.

File: src/cinematic.cs (hunk 17 above)
Comment (mandel-macaque) [reply src/cinematic.cs:452]: Same.

File: src/cinematic.cs (hunk 14 above)
Comment (mandel-macaque) [reply src/cinematic.cs:268]: Same.

File: src/cinematic.cs [hunk 23]
```diff
@@ -0,0 +1,494 @@
...
+		CGAffineTransform PreferredTransform { get; }
+
+		[Export ("quality")]
+		CNRenderingQuality Quality { get; }
+
+		[Export ("encodeRenderToCommandBuffer:frameAttributes:sourceImage:sourceDisparity:destinationImage:")]
+		bool EncodeRender (IMTLCommandBuffer commandBuffer, CNRenderingSessionFrameAttributes frameAttributes, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity, CVPixelBuffer destinationImage);
+
+		[Export ("encodeRenderToCommandBuffer:frameAttributes:sourceImage:sourceDisparity:destinationRGBA:")]
+		bool EncodeRender (IMTLCommandBuffer commandBuffer, CNRenderingSessionFrameAttributes frameAttributes, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity, IMTLTexture destinationRGBA);
```
Comment (tj-devel709) [inline src/cinematic.cs:173]:
```suggestion
		bool EncodeRender (IMTLCommandBuffer commandBuffer, CNRenderingSessionFrameAttributes frameAttributes, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity, IMTLTexture destinationRgba);
```
not quite sure if I remember the rule, but should this be as such?

**Review by tj-devel709 — APPROVED** [review]
Left a small thing to check, but other than that, LGTM

**Review by haritha-mohan — APPROVED** [review]

File: src/Cinematic/CNDecision.cs [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+#endif
+	public enum CNDecisionIdentifierType {
+		Single,
+		Group,
+	}
+
+	public partial class CNDecision {
+
+		public CNDecision (CMTime time, long detectionId, bool isStrong, CNDecisionIdentifierType identifierType)
+		{
```
Comment (rolfbjarne) [inline src/Cinematic/CNDecision.cs:28]: I believe you should call base with NSObjectFlag.Empty here:

```suggestion
		public CNDecision (CMTime time, long detectionId, bool isStrong, CNDecisionIdentifierType identifierType)
			: base (NSObjectFlag.Empty)
		{
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Pull request merged**